app/services/hyrax/collections/collection_member_query_service.rb

```diff
@@ -0,0 +1,91 @@
+module Hyrax
+  module Collections
+    # Responsible for retrieving collection members
+    class CollectionMemberQueryService
+      delegate :repository, to: :scope
+      attr_reader :scope, :params
+
+      # @param [Object] scope: Typically a controller object that responds to `repository`
+      # @param [ActionController::Parameters] query params
+      def initialize(scope:, params:)
```

Comment:
Can we pass `collection` in as an argument here rather than call `scope.collection`? This is one thing about the current way that really bugs me.  You'd have to pass `self` to the search builder then, rather than scope.

Reply:
I think I did what you intended here.